models/stg_stripe__pricing.sql

```diff
@@ -0,0 +1,114 @@
+{{ config(enabled=var('stripe__using_subscriptions', True)) }}
+
+{% if var('stripe__price', does_table_exist('price')) %}
+with pricing as (
+
+    select * 
+    from {{ ref('stg_stripe__pricing_tmp') }}
+),
+
+fields as (
+
+    select
+        {{
+            fivetran_utils.fill_staging_columns(
+                source_columns=adapter.get_columns_in_relation(ref('stg_stripe__pricing_tmp')),
+                staging_columns=get_pricing_columns()
+            )
+        }}
+
+        {{ fivetran_utils.source_relation(
+            union_schema_variable='stripe_union_schemas', 
+            union_database_variable='stripe_union_databases') 
+        }}
+
+    from pricing
+),
+
+final as (
+
+    select 
+        id as pricing_id,
+        active as is_active,
+        unit_amount,
+        currency,
+        recurring_interval,
+        recurring_interval_count,
+        recurring_usage_type,
+        recurring_aggregate_usage,
+        metadata,
+        nickname,
+        product_id,
+        billing_scheme,
+        created,
+        invoice_item_id,
+        is_deleted,
+        lookup_key,
+        tiers_mode,
+        transform_quantity_divide_by,
+        transform_quantity_round,
+        type as pricing_type,
+        source_relation
```

Comment:
Do you know if the fact that both versions have different fields will cause downstream implications? I have only done this solution where the two source tables essentially had the same fields.

Not to say this won't work, but worth thinking about how it will impact downstream models based on what the customer has.

Reply:
Yes I did take a look at how this model was originally getting brought downstream, and I believe the conclusion I had was that the fields being used are present in both `price` and `plan`, so we should be good. Only thing I made a note for was to check that the join that's being used still works when swapped to `pricing_id`
`invoice_line_item.plan_id = plan.plan_id`
I've been following the [stripe docs](https://stripe.com/docs/billing/migration/migrating-prices) which say how the migration to price affects invoices, so I reckon I'll give this a closer look when working on transforms